### DISCUSSION TITLE
api経由でデータアーカイブとzipデータのダウンロードは可能か？

### DISCUSSION THREAD

**Discussion by kohei-noda-qcrg** [growilabs/growi#7154] [Help / Questions] [answered]
## 概要
ドメインがexample.comだとした場合
https://example.com/_api/v3/
以下のapiを叩くことでデータのアーカイブ及びダウンロード
(https://example.com/admin/export で手動でできる操作)を行うことは可能でしょうか？

## 試したことと結果
https://docs.growi.org/redoc.html#operation/createExport
が該当のapiかと思い
https://example.com/_api/v3/export
を叩いてみたところ
{
    "ok": true
}
のレスポンスが返ってきて

https://docs.growi.org/redoc.html#operation/getExportStatus
の結果や
https://example.com/admin/export
のページを見る限りは新しいアーカイブファイルはできませんでした

Environment
------------

### Host

| item     | version |
| ---      | --- |
|OS        |Ubuntu 20.04.4 LTS|
|GROWI     |5.0.0|
|node.js   |16.14.2|
|npm       |8.5.0|
|Using Docker|yes|
|Using [growi-docker-compose][growi-docker-compose]|yes|

[growi-docker-compose]: https://github.com/weseek/growi-docker-compose

**Comment by yuki-takei** [comment, marked as answer]
長らくお返事ができず申し訳ありません。

[v7.5.0](https://github.com/growilabs/growi/releases/tag/v7.5.0) にて Enhanced Access Token（スコープ付き個人用アクセストークン）が実装され、API 経由でのアーカイブ操作が可能になりました。

**エクスポートの生成**
```
POST /api/v3/export
```
スコープ: `WRITE:ADMIN:EXPORT_DATA`

**エクスポート状態の確認**
```
GET /api/v3/export/status
```
スコープ: `READ:ADMIN:EXPORT_DATA`

**zip ファイルのダウンロード**
```
GET /admin/export/{fileName}
```
スコープ: `READ:ADMIN:EXPORT_DATA`

アクセストークンはユーザー設定画面から発行できます。

引き続きご不明な点がございましたら、お気軽に新しい Discussion を起票してください。
